packages/bpk-component-modal/src/BpkModalV2/BpkModal.tsx

```diff
@@ -0,0 +1,164 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ReactNode } from 'react';
+import { useEffect, useRef } from 'react';
+
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import BpkCloseButton from '../../../bpk-component-close-button';
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import BpkText, { TEXT_STYLES } from '../../../bpk-component-text';
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import { cssModules, withDefaultProps } from '../../../bpk-react-utils';
+
+import STYLES from './BpKModal.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export type Props = {
+  id: string | undefined;
+  ariaLabelledby: string;
+  children: ReactNode;
+  closeLabel: string;
+  fullScreenOnDesktop?: boolean;
+  isOpen: boolean;
+  noFullScreenOnMobile?: boolean;
+  onClose: () => void | null;
+  padded?: boolean;
+  title?: string | null;
+  wide?: boolean;
+};
+
+const Heading = withDefaultProps(BpkText, {
+  textStyle: TEXT_STYLES.label1,
+  tagName: 'h2',
+  className: getClassName('bpk-modal__title'),
+});
+
+type DialogProps = {
+  isDialogOpen: boolean;
+};
+
+const dialogSupported = typeof HTMLDialogElement === 'function';
```

Comment:
why do we need this check won't it always be the same type or does it differ between browser or depend on something else? 🙂

Reply:
Yes, this is used for the polyfill for Safari 14 and below.

Reply:
Yes, it is Safari 14 😞 cool, thanks for clarifying! can you maybe add a comment above to explain this? Maybe even as a TODO to remove this once we drop support for Safari 14!